### ISSUE TITLE
Feature Request: Ability to override generation help text

### ISSUE THREAD

**Issue by ty-i3** [l3nz/cli-matic#45]
This feature request adds optional functions that override the way cli-matic prints help, affording complete byte-by-byte control over the help text.

Rationale: Our internal CLI tooling needs to adhere to certain policies regarding help and usability for developers. Certain info such as Git build, version string, and grouping of commands needs to be printed in the prescribed format. This feature request would realize the ability to completely control the presentation of generated help text to satisfy those policies.

An example of generating global help:
```clojure
(def banner
  (str "Welcome to CLI-ABC123, Git "
       build-info/git-hash))

(defn commands-segment
  [cfg]
  (str "Commands:\n"
       (clojure.string/join
         "\n"
         (map
           #(format "   %-10s %s"
                    (get % :command)
                    (get % :description))
           (cfg :commands)))))

(defn my-global-help
  [cfg]
  (clojure.string/join "\n" [banner (commands-segment cfg) ...]))
```

and using it as the global help text generator:

```clojure
(run-cmd args {:app {:global-help my-global-help
                     ...}})
```

**Comment by l3nz** [comment]
So let me recap; this change adds optional functions that override the way text is output when printing help?

**Comment by ty-i3** [comment]
Yes, that's right!

**Comment by l3nz** [comment]
You'd do me a favor if you were to add an example under  `examples` that shows custom help generation. But I merged your changes and released as 0.1.18

**Comment by ty-i3** [comment]
Done as requested.
